aten/src/ATen/cuda/CUDAGraphsUtils.cuh

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <ATen/CUDAGeneratorImpl.h>
+#include <ATen/cuda/CUDAEvent.h>
+#include <c10/core/StreamGuard.h>
+#include <c10/cuda/CUDAGuard.h>
+
+namespace at {
+namespace cuda {
+namespace philox {
+
+// We can't write a __device__ function in CUDAGeneratorImpl.h, because it's in ATen.
+// Also, whatever call unpacks PhiloxCudaState in consumer kernels must be inlineable.
+// Easiest thing that comes to mind is, define a free function here, in ATen/cuda.
+// Any cuda consumer can include this header.
+__device__ __forceinline__ std::tuple<uint64_t, uint64_t>
+unpack(at::PhiloxCudaState arg) {
+  if (arg.captured_) {
+    return std::make_tuple(arg.seed_, *(arg.offset_.ptr) + arg.offset_intragraph_);
+  } else {
+    return std::make_tuple(arg.seed_, arg.offset_.val);
```

Comment:
Look ma, no non-capture overhead! 🤣

Reply:
fishing for compliments?? what can i say, your idea was awesome =D